NeoML/src/Dnn/Layers/TiedEmbeddingsLayer.cpp

```diff
@@ -36,29 +36,35 @@ void CTiedEmbeddingsLayer::SetChannelIndex( int val )
 	channelIndex = val;
 }
 
-static const int CnnTiedEmbeddingsLayerVersion = 2000;
+static const int CnnTiedEmbeddingsLayerVersion = 2001;
 
 void CTiedEmbeddingsLayer::Serialize( CArchive& archive )
 {
-	archive.SerializeVersion( CnnTiedEmbeddingsLayerVersion, CDnn::ArchiveMinSupportedVersion );
+	int version = archive.SerializeVersion(CnnTiedEmbeddingsLayerVersion, CDnn::ArchiveMinSupportedVersion);
 	CBaseLayer::Serialize( archive );
 
-	archive.Serialize( embeddingsLayerName );
+	if( version < 2001 ) {
+		if( archive.IsLoading() ) {
+			CString embeddingLayerName;
+			archive.Serialize( embeddingLayerName );
+			embeddingPath = { embeddingLayerName };
+		} else {
+			archive.Serialize( embeddingPath.Last() );
```

Comment:
By storing serialize always the embeddingPath.
Check for a version only for the loading.